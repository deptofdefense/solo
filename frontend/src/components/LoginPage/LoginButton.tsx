import React from "react";
import classes from "./LoginPage.module.scss";

interface LoginButtonProps {
  onClick: React.MouseEventHandler;
}

const LoginButton: React.FC<LoginButtonProps> = ({ onClick }) => (
  <div className={classes.btnRoot}>
    <button className="usa-button usa-button--big" onClick={onClick}>
      CAC Login
    </button>
  </div>
);

export default LoginButton;
