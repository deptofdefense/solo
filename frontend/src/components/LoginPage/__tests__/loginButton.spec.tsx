import React from "react";
import { render, fireEvent } from "test-utils";
import LoginButton from "../LoginButton";

describe("Login Button", () => {
  const clickHandler = jest.fn();

  afterEach(() => {
    clickHandler.mockReset();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<LoginButton onClick={clickHandler} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("calls click handler when clicked", async () => {
    const { getByText } = render(<LoginButton onClick={clickHandler} />);
    const button = getByText(/CAC Login/);
    fireEvent.click(button);
    expect(clickHandler).toHaveBeenCalledTimes(1);
  });
});
