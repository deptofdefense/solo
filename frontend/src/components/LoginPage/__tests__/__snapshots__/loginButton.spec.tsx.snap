// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Button matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="btnRoot"
  >
    <button
      class="usa-button usa-button--big"
    >
      CAC Login
    </button>
  </div>
</DocumentFragment>
`;
